### PR TITLE
Add `tenants open` and `apis open` [CLI-101] [CLI-133]

### PR DESCRIPTION
**Pull request by Widcket**
### Description

This PR adds the following commands:
- `tenants open` that opens the tenant's settings page on the Dashboard. It takes the tenant's domain as a parameter.
- `apis open` that opens the API's settings page on the Dashboard. It takes either the API Id or audience as a parameter.

This is important as it provides a handy escape hatch, because the CLI does not support every parameter available for APIs, and does not allow to configure tenants ATM.

This PR also fixes a bug that caused the open logic to not work with PUS1 tenants, that do not have the region in the URL.

### Testing

- [ ] This change adds test coverage for new/changed/fixed functionality

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR or in auth0.com/docs
- [X] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used, if not `master`

### CODE REVIEW

File: internal/cli/apis.go [hunk 1]
```diff
@@ -324,6 +326,52 @@ auth0 apis delete <id>`,
...
+				err := apiID.Pick(cmd, &inputs.ID, cli.apiPickerOptions)
+				if err != nil {
+					return err
+				}
+			} else {
+				inputs.ID = args[0]
+			}
+
+			// Heuristics to determine if this a valid ID, or an audience value
+			if _, err := url.ParseRequestURI(inputs.ID); err == nil || len(inputs.ID) != 24 {
```
Comment (jfatta) [inline internal/cli/apis.go:354]: what's `24`coming from ? is inputs.ID a url? in that case I wonder if we can make it more explicit on the command help

File: internal/cli/apis.go [hunk 2]
```diff
@@ -324,6 +326,52 @@ auth0 apis delete <id>`,
...
+				if err := ansi.Waiting(func() error {
+					api, err := cli.api.ResourceServer.Read(url.PathEscape(inputs.ID))
+					if err != nil {
+						return err
+					}
+					inputs.ID = auth0.StringValue(api.ID)
+					return nil
+				}); err != nil {
+					return fmt.Errorf("An unexpected error occurred while trying to get the API Id for '%s': %w", inputs.ID, err)
```
Comment (jfatta) [inline internal/cli/apis.go:363]: NAB: I know we're doing this in other places already, but I'd suggest starting error messages with lowercase (as well as avoiding ending punctuation). More context here: https://github.com/golang/go/wiki/CodeReviewComments#error-strings

File: internal/cli/apps.go [hunk 3]
```diff
@@ -669,7 +667,7 @@ func openAppCmd(cli *cli) *cobra.Command {
 		Args:    cobra.MaximumNArgs(1),
 		Short:   "Open application settings page in Auth0 Manage",
 		Long:    "Open application settings page in Auth0 Manage.",
-		Example: `auth0 apps open <id>`,
+		Example: "auth0 apps open <id>",
```
Comment (jfatta) [inline internal/cli/apps.go:670]: from prev comment, If id could adopt different forms (I'm not sure) I'd suggest including some placeholders here in the example section.

File: internal/cli/tenants.go [hunk 4]
```diff
@@ -93,3 +101,67 @@ func useTenantCmd(cli *cli) *cobra.Command {
...
+		},
+	}
+
+	return cmd
+}
+
+func (c *cli) tenantPickerOptions() (pickerOptions, error) {
+	tens, err := c.listTenants()
+	if err != nil {
+		return nil, fmt.Errorf("Unable to load tenants due to an unexpected error: %w", err)
```
Comment (jfatta) [inline internal/cli/tenants.go:144]: ditto

File: internal/cli/tenants.go [hunk 5]
```diff
@@ -93,3 +101,67 @@ func useTenantCmd(cli *cli) *cobra.Command {
...
+	}
+
+	var (
+		priorityOpts, opts pickerOptions
+	)
```
Comment (jfatta) [inline internal/cli/tenants.go:149]:
```suggestion
          var priorityOpts, opts pickerOptions
```

**Review by jfatta — APPROVED** [review]
looks good. minor suggestions inline.
thanks

File: internal/cli/apis.go (hunk 1 above)
Comment (Widcket) [reply internal/cli/apis.go:354]: 24 is the string length of the ID. Audiences are usually URLs, but not necessarily, so there is that additional heuristic to help disambiguate. I'll make it more explicit on the command help and add a comment explaining the values, thanks!

File: internal/cli/tenants.go (hunk 4 above)
Comment (Widcket) [reply internal/cli/tenants.go:144]: Same as above.

File: internal/cli/apps.go (hunk 3 above)
Comment (Widcket) [reply internal/cli/apps.go:670]: I went with `auth0 apps open <id|audience>` in https://github.com/auth0/auth0-cli/pull/253/commits/62c64d535f533809886487d159ef00c33b04881b

File: internal/cli/apis.go (hunk 1 above)
Comment (Widcket) [reply internal/cli/apis.go:360]: Done in https://github.com/auth0/auth0-cli/pull/253/commits/62c64d535f533809886487d159ef00c33b04881b

**Pull request merged**